### PR TITLE
Sign In Details

### PR DESCRIPTION
**Pull request by michael-grace**
Not the best way of implementing, but definitely the simplest. Not on any menus, cause in true MyRadio fashion, We ❤️️ Hidden Links (/Scheduler/tracking)

### CODE REVIEW

**Comment by URY-GitHub** [comment]
Thank you for this pull request, @michael-grace!

@SplashyFish14, would you mind reviewing this PR, please? Feel free to ask anyone for help if you're not quite sure what you're doing.

**Comment by michael-grace** [comment]
could possibly be done with a selection thing to select what locations you're interested in viewing

**Comment by markspolakovs** [comment]
@michael-grace that was exactly what I was thinking.

Also, consider filtering by time, instead of pulling up every single timeslot in the term? It may work initially, but with more studio shows it may become unworkable.

**Review by markspolakovs — APPROVED** [review]
Looks good for a MVP, although see comment above

**Comment by markspolakovs** [comment]
Also, maybe add it to menu? It won't show up without appropriate perms anyway, so no point in hiding it.

**Comment by markspolakovs** [comment]
image

You  may need to make `guest_info` HTML and add `<br>`s, as the form says "one on each line". (If you do that, be careful about XSS escaping.)

**Comment by michael-grace** [comment]
image

File: src/Classes/ServiceAPI/MyRadio_Timeslot.php [hunk 1]
```diff
@@ -1261,7 +1261,7 @@ public function signInGuests(string $guestInfo, $locationid)
         self::$db->query(
             'INSERT INTO sis2.guest_signin (show_season_timeslot_id, signerid, location, guest_info)
                 VALUES ($1, $2, $3, $4)',
-            [$this->getID(), MyRadio_User::getInstance()->getID(), $locationid, $guestInfo]
+            [$this->getID(), MyRadio_User::getInstance()->getID(), $locationid, htmlspecialchars($guestInfo, ENT_QUOTES)]
```
Comment (markspolakovs) [inline src/Classes/ServiceAPI/MyRadio_Timeslot.php:1264]: Screm - I'd rather put in the cleanest data possible, and format it when you display it. (Also, Twig might take care of escaping itself?)

Comment (markspolakovs) [reply src/Classes/ServiceAPI/MyRadio_Timeslot.php:1264]: Wait, you use `html`, nvm - either way, move `htmlspecialchars` to Scheduler/tracking.php

Comment (michael-grace) [reply src/Classes/ServiceAPI/MyRadio_Timeslot.php:1264]: nah, this seems to be the only way which works, remember say on quotes (rip) and show descriptions you can xss

**Pull request merged**